tests/test_DGEList.py

```diff
@@ -36,6 +36,26 @@ def test_setting_DGElist_counts():
     with pytest.raises(ValueError):
         dge_list().counts = np.array(1, 1, np.nan)
 
+def testing_setting_samples_and_counts():
```

Comment:
These look good for now! Here are two things to consider when making future unit tests:

- Always make an assertion dependent on the action that you made (`py.test` looks for the assertion syntax too)
- Create a minimal example. For instance, does `to_remove_zeroes` change the behavior of any of the tests? If not, it may not be needed.

Here is some pseudo-code addressing point (1):

```python
# program.py
class dud:
    def __init__(self, value):
        self.value = value
    def return_value(self):
        return self.value
```

```python
# test_program.py
from program import dud

def test_return_value():
    actual = 2
    expected = dud(actual).return_value()
    assert expected == actual
```

For some real-life examples check out this test suite on a class I created in an open source project:

https://github.com/clintval/sample-sheet/blob/master/tests/test_sample_sheet.py
    
Let's keep these tests... I'll go through the entire `edgePy` suite in a few days and update all tests for proper coverage. Then it will be a useful foundation to reference for future feature requests.

Reply:
As an aside, it is extremly common for test suites to have more lines of code than the functions or classes they are testing.

Using [parameterization](https://docs.pytest.org/en/latest/parametrize.html) and [fixtures](https://docs.pytest.org/en/latest/fixture.html) greatly reduces testing lines of code so I recommend reading up on those topics if you are interested on more efficient unit test writing.

For example, you could collapse all of the testing you wrote to just:

```python
import pytest

from numpy import array_equal

@pytest.mark.parametrize("samples", [None, np.array(['1','2','3'])])
@pytest.mark.parametrize("counts", [None, np.array([[2, 2, 2], [2, 2, 2], [2, 2, 2]])])
def test_sample_counts_validation(samples, counts):
    dge_list = DGEList(samples=samples, counts=counts)
    assert array_equal(dge_list.samples, samples)
    assert array_equal(dge_list.counts, counts)

Reply:
^ The parameterization code above may not handle the `None` cases but you get the idea.

Reply:
I didn't know about parameterization and fixtures, so that's super helpful! And what you wrote just checks the cases that don't error, right? Even though there is a slight problem I think, since anytime counts=None, I think I need to_remove_zeros = False. But I'll check to make sure.